src/abstractdataframe/prettytables.jl

```diff
@@ -65,24 +65,31 @@ end
 
 function _pretty_tables_float_formatter(v, i::Integer, j::Integer,
                                         float_cols::Vector{Int},
-                                        indices::Vector{Vector{Int}},
-                                        padding::Vector{Vector{Int}},
+                                        align_col::Vector{Int},
                                         compact_printing::Bool)
     isempty(float_cols) && return v
 
     # We apply this formatting only to the columns that contains only floats.
     ind_col = findfirst(==(j), float_cols)
 
     if ind_col !== nothing
-        ind_row = findfirst(==(i), indices[ind_col])
+        align_col_i = align_col[ind_col]
 
-        if ind_row !== nothing
-            pad = padding[ind_col][ind_row]
+        # Convert the value to text.
+        str = sprint(print, v, context = :compact => compact_printing)
 
-            # Return the formatted number.
-            str = sprint(print, v, context = :compact => compact_printing)
-            return " "^pad * str
-        end
+        # We want to align everything at '.'.
+        id_dp = findfirst('.', str)
+
+        # Compute the require padding to align the cell.
+        pad = id_dp !== nothing ? align_col_i - id_dp : align_col_i - 1
+
+        # Pad cannot be negative.
+        # NOTE: This is just a failsafe, `pad` will be negative only if we have
```

Comment:
This is a nice safeguard. I understand that already `id_dp` is `nothing` only if we have a bug - right?

Reply:
No. I can be nothing at least in two occasions:

1. The value is `missing`.
2. The user defines a structure that is `<: AbstractFloat` in which the printing method does not contain a decimal point.